core/src/main/java/org/openedx/core/ui/IAPUI.kt

```diff
@@ -0,0 +1,326 @@
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import org.openedx.core.R
+import org.openedx.core.ui.theme.appColors
+import org.openedx.core.ui.theme.appShapes
+import org.openedx.core.ui.theme.appTypography
+
+@Composable
+fun ValuePropUpgradeFeatures(modifier: Modifier = Modifier, courseName: String) {
+    Column(
+        modifier = modifier
+            .padding(all = 16.dp)
+    ) {
+        Text(
+            modifier = Modifier.padding(vertical = 32.dp),
+            text = stringResource(
+                id = R.string.iap_upgrade_course,
+                courseName
+            ),
+            color = MaterialTheme.appColors.textPrimary,
+            style = MaterialTheme.appTypography.titleLarge,
+            maxLines = 3,
+            overflow = TextOverflow.Ellipsis,
+        )
+        CheckmarkView(stringResource(id = R.string.iap_earn_certificate))
+        CheckmarkView(stringResource(id = R.string.iap_unlock_access))
+        CheckmarkView(stringResource(id = R.string.iap_full_access_course))
+    }
+}
+
+@Composable
+fun CheckmarkView(text: String) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(vertical = 16.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Icon(
+            modifier = Modifier.padding(end = 16.dp),
+            imageVector = Icons.Filled.Check,
+            contentDescription = null,
+            tint = MaterialTheme.appColors.certificateForeground
+        )
+        Text(
+            modifier = Modifier.weight(1f),
+            text = text,
+            color = MaterialTheme.appColors.textPrimary,
+            style = MaterialTheme.appTypography.labelLarge
+        )
+    }
+}
+
+@Composable
+fun PriceLoadErrorDialog(onConfirm: () -> Unit, onDismiss: () -> Unit) {
+    UpgradeErrorDialog(
+        title = stringResource(id = R.string.iap_error_title),
+        description = stringResource(id = R.string.iap_error_price_not_fetched),
+        confirmText = stringResource(id = R.string.iap_label_refresh_now),
+        onConfirm = onConfirm,
+        dismissText = stringResource(id = R.string.core_cancel),
+        onDismiss = onDismiss
+    )
+}
+
+@Composable
+fun CourseAlreadyPurchasedErrorDialog(
+    onRefresh: () -> Unit,
+    onGetHelp: () -> Unit,
+    onDismiss: () -> Unit
+) {
+    AlertDialog(
+        title = {
```

Comment:
Please add these to all the dialogs:

```
        modifier = Modifier
            .background(
                color = MaterialTheme.appColors.background,
                shape = MaterialTheme.appShapes.cardShape
            )
            .padding(bottom = 8.dp),
        shape = MaterialTheme.appShapes.cardShape,
        backgroundColor = MaterialTheme.appColors.background,
```        
And this to Titles:

```
        color = MaterialTheme.appColors.textPrimary,
        style = MaterialTheme.appTypography.titleMedium,        
```

And this to Description:
```
        color = MaterialTheme.appColors.textPrimary,
        style = MaterialTheme.appTypography.bodyMedium